File: python/middle/minimum_move_to_equal_array_elements_ii.py
class Solution(object):
    def minMoves2(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0 or len(nums) == 1:
            return 0
        start = 0
        end = len(nums)-1
        ret = 0
        nums.sort()
        while start < end:
            ret += (nums[end]-nums[start])
            start += 1
            end -= 1
        return ret
        

File: cpp/middle/minimum_move_to_equal_array_elements_ii.cpp
class Solution {
public:
    int minMoves2(vector<int>& nums) {
        if(nums.size() == 0 || nums.size() == 1)
            return 0;
        sort(nums.begin(), nums.end());
        int i = 0;
        int j = nums.size()-1;
        int ret = 0;
        while(i < j)
        {
            ret += nums[j] - nums[i];
            ++i;
            --j;
        }
        return ret;
    }
};


Conclusion:

a middle level algorithm issue, we just need to sort the array and caculate the difference bettwen the latest one and the first one, then move
and caculate...


Time Complexity: O(lgN)
Space Complexity: O(1)
